ICAFIX/scripts/computeRecleanFeatures.m

```diff
@@ -0,0 +1,266 @@
+            NewFeatures(k,47)=mean(abs(CIFTI.cdata(:,k)))./mean(abs(Volume_data(NonGrey==1,k))); %CIFTI/NonGrey
+            NewFeatures(k,48)=mean(abs(CIFTI.cdata(CORTEX_idx_list,k)))./mean(abs(Volume_data(NonGrey==1,k))); %Cerebral Cortex/NonGrey
+            NewFeatures(k,49)=mean(abs(CIFTI.cdata(:,k)))./mean(abs(Volume_data(NonGreyPlusEdge==1,k))); %CIFTI/NonGreyPlusEdge
+            NewFeatures(k,50)=mean(abs(CIFTI.cdata(CORTEX_idx_list,k)))./mean(abs(Volume_data(NonGreyPlusEdge==1,k))); %Cerebral Cortex/NonGreyPlusEdge
+            NewFeatures(k,51)=mean(abs(CIFTIMSMAllSmooth.cdata(:,k)))./mean(abs(VolumeSmooth_data(NonGrey==1,k))); %CIFTI/NonGrey
+            NewFeatures(k,52)=mean(abs(CIFTIMSMAllSmooth.cdata(CORTEX_idx_list,k)))./mean(abs(VolumeSmooth_data(NonGrey==1,k))); %Cerebral Cortex/NonGrey
+            NewFeatures(k,53)=mean(abs(CIFTIMSMAllSmooth.cdata(:,k)))./mean(abs(VolumeSmooth_data(NonGreyPlusEdge==1,k))); %CIFTI/NonGreyPlusEdge
+            NewFeatures(k,54)=mean(abs(CIFTIMSMAllSmooth.cdata(CORTEX_idx_list,k)))./mean(abs(VolumeSmooth_data(NonGreyPlusEdge==1,k))); %Cerebral Cortex/NonGreyPlusEdge
+
+            for idx=1:max(CorticalParcellation.cdata)
+                NewFeatures(k,idx+54)=mean(abs(CIFTIMSMAll.cdata(CorticalParcellation.cdata==idx,k)));
+            end
+
+            for idx=1:size(SubRegionsROI.cdata, 2)
+                NewFeatures(k,idx+54+max(CorticalParcellation.cdata))=mean(abs(CIFTIMSMAll.cdata(SubRegionsROI.cdata(:,idx)==1,k)));
+            end
+
+        end
+        save([SubjFolderlist '/MNINonLinear/Results/' fMRIName '/' fMRIName '_hp' hp '.ica/original_fix_features.mat'], 'OrigFeatures', '-v7.3')
+        save([SubjFolderlist '/MNINonLinear/Results/' fMRIName '/' fMRIName '_hp' hp '.ica/reclean_features.mat'], 'NewFeatures', '-v7.3')
+        save([SubjFolderlist '/MNINonLinear/Results/' fMRIName '/' fMRIName '_hp' hp '.ica/fix_prob.mat'], 'probs', '-v7.3')
+
+        T_fix = cell2table(num2cell(OrigFeatures), 'RowNames', row_names);
+        T_reclean = cell2table(num2cell(NewFeatures), 'RowNames', row_names);
+        T_fix_prob = cell2table(num2cell(probs), 'RowNames', row_names);
+        T_fix_reclean = cell2table(num2cell([OrigFeatures, NewFeatures]), 'RowNames', row_names);
+
+        writetable(T_fix, [SubjFolderlist '/MNINonLinear/Results/' fMRIName '/' fMRIName '_hp' hp '.ica/original_fix_features.csv'], 'WriteRowNames', true);
+        writetable(T_reclean, [SubjFolderlist '/MNINonLinear/Results/' fMRIName '/' fMRIName '_hp' hp '.ica/reclean_features.csv'], 'WriteRowNames', true);
+        writetable(T_fix_reclean, [SubjFolderlist '/MNINonLinear/Results/' fMRIName '/' fMRIName '_hp' hp '.ica/fix_reclean_features.csv'], 'WriteRowNames', true);
+        writetable(T_fix_prob, [SubjFolderlist '/MNINonLinear/Results/' fMRIName '/' fMRIName '_hp' hp '.ica/fix_prob.csv'], 'WriteRowNames', true);
```

Comment:
Why are we making 2 additional copies of the original features (which don't appear to otherwise be used by this code)?  Why 2 formats of the new features, and...looks like original fix feature probabilities, which this code also doesn't appear to use other than writing it back out?

Reply:
These files are helpful for me to debug in either matlab or python environment.

Reply:
And FIX also save features in both formats.